### DISCUSSION TITLE
Renaming function stops it from working

### DISCUSSION THREAD

**Discussion by TheBurgerCoder** [phcode-dev/phoenix#1131] [Q&A]
`<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <title>Title</title>
        <link rel="stylesheet" href="styles.css" />
        <script src="script.js"></script>
    </head>
    <body>
        <h1 class="title">Hello World</h1>
        <div id="mainDiv">Hello Div</div>
        <br/>
        <button class="arrowkeys" id="left" onclick="buttonClicked()">&lArr;</button>
        <button class="arrowkeys" id="right" onclick="buttonClick()">&rArr;</button>
        
    </body>
</html>
`
`let wall = 0;
let rightbtn = document.getElementById("right");
function buttonClicked() {
    wall =+ 1;
    document.getElementById("mainDiv").textContent = "Button Clicked times: " + wall;
}
function buttonClick() {
wall =+ 1;
    document.getElementById("mainDiv").textContent = "Button Clicked times: " + wall;
}

`
Why does the left button work but no matter what method I try the right one won't do the same I literally just renamed it?

**Comment by abose** [comment]
From the provided code, there are a few issues and things to address:

1. **HTML Structure Issues**:
    - The `<body>` tag is closing, but there's no opening tag.
    - `<title>` is typically used inside the `<head>` tag, and you haven't provided a starting `<head>` or `<html>` tag.
    - There seems to be a missing definition for the buttons and the mainDiv that your JavaScript is referencing.

2. **JavaScript Issue**:
    - In the function `buttonClicked()` and `buttonClick()`, you're using `=+` which is not a valid assignment operator. You probably meant `+=`, which increments the `wall` variable by the right operand. With `=+`, you're effectively assigning the value 1 to `wall` every time the button is clicked.

3. **Missing Information**: You mentioned a left button and a right button, but the provided code doesn't contain these buttons' definitions. It's also unclear how you've linked each button with the respective functions.

Here's a basic correction:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Title</title>
</head>
<body>
    <button id="left" onclick="buttonClicked()">Left Button</button>
    <button id="right" onclick="buttonClick()">Right Button</button>
    <div id="mainDiv"></div>
    
    <script>
        let wall = 0;
        
        function buttonClicked() {
            wall += 1;
            document.getElementById("mainDiv").textContent = "Button Clicked times: " + wall;
        }
        
        function buttonClick() {
            wall += 1;
            document.getElementById("mainDiv").textContent = "Button Clicked times: " + wall;
        }
    </script>
</body>
</html>
```

In this corrected code, both the left and right buttons should increment the `wall` variable by 1 when clicked. If one button is not working as expected, it's likely due to some other external issue or a mistake not shown in the provided code.

**Comment by abose** [comment]
Please use the discord for programming questions like these 
https://discord.com/channels/879126209114542181/1026109472042668123

Closing as this is not a product specific issue
